app/models/spree/retail/shopify/generate_pos_order.rb

```diff
@@ -136,6 +137,12 @@ def transition_order_from_confirm_to_complete!(order)
           mark_as_shipped(order)
         end
 
+        def apply_adjustment(spree_order, shopify_order)
+          shopify_discount = shopify_order.discount_codes.first
+          return unless shopify_discount
+          spree_order.adjustments.create!(amount: shopify_discount.amount, label: shopify_discount.code, order: spree_order)
```

Comment:
do you think it's worth considering `total_discount` on the shopify order? i'm not entirely sure if the shop girls ever create more than one discount on the orders or how orders are made "free"